contracts/FeeCustody.sol

```diff
@@ -0,0 +1,347 @@
+        require(IChainlink(oracles[_asset]).decimals() == 8, "!ASSET/USD");
+        require(_pathLen < 2, "invalid intermediary path");
+        require(_swapFeeLen > 0 && _swapFeeLen < 2, "invalid pool fees array length");
+
+        // If not set asset
+        if(!_assetExists){
+          assets[lastAssetIdx] = _asset;
+          ++lastAssetIdx;
+          assetExists[_asset] = true;
+        }
+
+        // Set oracle for asset
+        oracles[_asset] = _oracle;
+
+        // Multiple pool swaps are encoded through bytes called a `path`.
+        // A path is a sequence of token addresses and poolFees that define the pools used in the swaps.
+        // The format for pool encoding is (tokenIn, fee, tokenOut/tokenIn, fee, tokenOut)
+        // where tokenIn/tokenOut parameter is the shared token across the pools.
+        if(_pathLen > 0){
+          intermediaryPath[_asset] = abi.encodePacked(_asset, _poolFees[0], _intermediaryPath[0], _poolFees[1], address(distributionToken));
+        }else{
+          intermediaryPath[_asset] = abi.encodePacked(_asset, _poolFees[0], address(distributionToken));
+        }
+
+        emit NewAsset(_asset, intermediaryPath[_asset]);
+    }
+
+    /**
+     * @notice
+     * recover all assets
+     * @dev Can be called by admin
+     */
+    function recoverAllAssets() external onlyOwner {
+        // For all added assets, if not removed, send to protocol revenue recipient
+        for(uint i = 0; i < lastAssetIdx; i++){
+          _recoverAsset(assets[i]);
+        }
+    }
+
+    /**
+     * @notice
+     * recover specific asset
+     * @dev Can be called by admin
+     * @param _asset asset to recover
+     */
+    function recoverAsset(address _asset) external onlyOwner {
+      require(_asset != address(0), "!address(0)");
+      _recoverAsset(_asset);
+    }
+
+    /**
+     * @notice
+     * recovers asset logic
+     * @param _asset asset to recover
+     */
+    function _recoverAsset(address _asset) internal {
+      IERC20 asset = IERC20(_asset);
+      uint256 bal = asset.balanceOf(address(this));
+      if(bal > 0){
+        asset.transfer(protocolRevenueRecipient, bal);
+        emit RecoveredAsset(_asset);
+      }
+    }
+
+    /**
+     * @notice
+     * set fee distributor
+     * @dev Can be called by admin
+     * @param _feeDistributor new fee distributor
+     */
+    function setFeeDistributor(address _feeDistributor) external onlyOwner {
+        require(_feeDistributor != address(0), "!address(0)");
+        feeDistributor = IFeeDistributor(_feeDistributor);
+        emit NewFeeDistributor(_feeDistributor);
+    }
+
+    /**
+     * @notice
+     * set rbn locker allocation pct
+     * @dev Can be called by admin
+     * @param _pctAllocationForRBNLockers new allocation for rbn lockers
+     */
+    function setRBNLockerAllocPCT(uint256 _pctAllocationForRBNLockers) external {
+        require(_pctAllocationForRBNLockers != 0, "!0");
```

Comment:
Is there a situation where we would want to set the `pctAllocationForRBNLockers` to be 0?

Reply:
good call